driver-core/src/test/unit/com/mongodb/operation/AsyncQueryBatchCursorSpecification.groovy

```diff
@@ -453,33 +453,71 @@ class AsyncQueryBatchCursorSpecification extends Specification {
                 cursor.close()
                 it[6].onResult(response, null)
             }
-            1 * connectionB.commandAsync(NAMESPACE.databaseName, createKillCursorsDocument(initialResult.cursor), _, primary(),
-                    _, _, _) >> {
-                it[6].onResult(null, null)
-            }
         } else {
             1 * connectionA.getMoreAsync(_, _, _, _, _) >> {
                 // Simulate the user calling close while the getMore is in flight
                 cursor.close()
                 it[4].onResult(response, null)
             }
-            1 * connectionB.killCursorAsync(NAMESPACE, [initialResult.cursor.id], _) >> {
-                it[2].onResult(null, null)
-            }
         }
 
         then:
-        thrown(MongoException)
+        noExceptionThrown()
 
         then:
         connectionA.getCount() == 0
-        connectionB.getCount() == 0
         connectionSource.getCount() == 0
+        cursor.isClosed()
 
         where:
         serverVersion                | commandAsync         | response
-        new ServerVersion([3, 2, 0]) | true                 | documentResponse([])
-        new ServerVersion([3, 0, 0]) | false                | new QueryResult(NAMESPACE, [], 42, SERVER_ADDRESS)
+        new ServerVersion([3, 2, 0]) | true                 | documentResponse([], 0)
```

Comment:
Can we keep the tests where the response includes a non-zero cursor id and add a new test where it's 0?